### PR TITLE
POK-4-DX Tooling

### LINKED ISSUE #4: POK-4- 🧰 Tooling - chore: Add DX Tooling to Application
# Add DX Tooling to Application

Adding good developer experience (DX) tooling is essential for any robust, production ready application. Please add the tools listed below to the app.

## DX Tools
- eslint
- prettier

### ESLint
1. Add `eslint`
```bash
npm install eslint --save-dev
```
2. Add `eslint` config file
```
npx eslint --init
```
This step will walk you through several questions. Answer like this:

Image

In the root directory, create a new file `.eslintignore`:
```bash
touch .eslintignore
```

Add the following to that file:
```
dist
```

Finally, update `package.json` with the following:

```json
{
"scripts":{
 "lint": "eslint --ignore-path .eslintignore --ext .js, .",
 "lint:fix": "eslint --ignore-path .eslintignore --ext .js, . --fix" 
 }
}
```

### Prettier
1. Add Prettier
```bash
npm install --save-dev --save-exact prettier
```
2. Configure Prettier
```bash
echo {"singleQuote": true}> .prettierrc.json
```
3. Add `.prettierignore` file
```bash
touch .prettierignore
```
Add the following:
```
// .prettierignore

# Ignore artifacts:
dist
coverage
```
4. In the `eslint` config file, add the following:
```
"extends": [
...,
"prettier"
]
```

## Guidance
[Adding tools to React / Parcel App](https://medium.com/@narendersaini32/setup-reactjs-with-parceljs-eslint-less-and-babel-in-5-min-2020-8401b6cab1ad)
[Prettier Setup](https://prettier.io/docs/en/install.html)
[Commit Better Code with Husky, Prettier, ESLint, and Lint-Staged](https://www.coffeeclass.io/articles/commit-better-code-with-husky-prettier-eslint-lint-staged)

### PR DESCRIPTION
**Pull request by mathiuskitchens**
# Description

Added DX tools: `eslint` and `prettier`

## Type of change

- [ ] - fix: - patches a bug in the codebase
- [ ] - feat: - introduces a new feature
- [ ] - build: - relates to build tooling / project scaffolding
- [x] - chore: - non-code related changes, like adding packages / libraries
- [ ] - docs: - changes related to documentation

## Related Issue

Does this change have a related issue in the project?

- [x] - Yes - Closes #4 
- [ ] - No

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "echo": {"singleQutore": true}
```
Comment (wijohnst) [inline .prettierrc.json]: The `echo` command is a zsh/bash command and was meant to be run from the command line. The result would be that this `JSON` would be added to the `.prettierrc.json` file"

```suggestion
   {
     "singleQuote": true
   }
```

**Review by wijohnst — COMMENTED** [review]
Left a suggestion for updating the prettier config. 

Also, how did you confirm that eslint and prettier are working?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,9 @@
   "description": "A Pokédex implemented in HTML, CSS, and vanilla JavaScript",
   "main": "index.js",
   "scripts": {
-    "start": "parcel src/index.html"
+    "start": "parcel src/index.html",
+    "lint": "eslint --ignore-path .eslintignore --ext .js, .",
+    "lint:fix": "eslint --ignore-path .eslintignore --ext .js, . --fix"
```
Comment (wijohnst) [inline package.json]: Can we also add a "prettier" command that runs prettier against all files in the app?

**Comment by mathiuskitchens** [comment]
> Left a suggestion for updating the prettier config.
> 
> Also, how did you confirm that eslint and prettier are working?

I made changes to make formatting bad and they fixed it. Is that enough? or is there a real test?

File: src/index.js [hunk 3]
```diff
@@ -1 +1,5 @@
 console.log("Search a Pokemon");
+
+//test hello
```
Comment (wijohnst) [inline src/index.js]: Can be removed

**Review by wijohnst — APPROVED** [review]
:shipit:

**Pull request merged**